feat: add monitoring for users across requests

robrap opened the pull request:
## Description

We already add the user id (imperfectly) to many requests.
However, when a user starts off unauthenticated, it is not
possible to correlate to those requests. Adding the raw
IP chain provides that possibility. See new custom attribute
ip_chain.raw.

robrap committed:
feat: add monitoring for users across requests

We already add the user id (imperfectly) to many requests.
However, when a user starts off unauthenticated, it is not
possible to correlate to those requests. Adding the raw
IP chain provides that possibility. See new custom attribute
ip_chain.raw.

robrap committed:
fixup! fix quality

timmc-edx commented on openedx/core/lib/x_forwarded_for/middleware.py:
```diff
@@ -51,6 +51,12 @@ def process_request(self, request):
+            # .. custom_attribute_description: The actual contents of the raw IP chain. Could
+            #      be used to correlate authenticated and unauthenticated requests for the same
+            #      user.
+            set_custom_attribute('ip_chain.raw', ip_chain)
```
Do we need to format this list to a string, or is `set_custom_attribute` going to take care of it for us?

timmc-edx replied:
(Adding a test would be great, and probably quick as well.)

robrap replied:
1. Added a test.
2. Added formatting.

robrap committed:
fixup! add test and formatting

Test new custom attribute, and add
`, ` formatting to make it more human
readable.
